modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/FileUtil.java

```diff
@@ -236,13 +237,74 @@ public static Path getFile(Path dirPath, String glob, String regex)
 	}
 
 	public static File getJarFile(Class<?> clazz) throws Exception {
-		ProtectionDomain protectionDomain = clazz.getProtectionDomain();
+		if (clazz == null) {
```

Comment:
This is a confusing non-obvious way to use this API versus the other 'getArchetypeFile' api.  can you unify them?

Reply:
In other words `FileUtil.getJarFile(clazz)` no longer is a good reasonable name.  I think we need to move everything to the ProjectTemplatesUtil.getArchetypeFile() or something similar.